src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -5,155 +5,200 @@
 import by.custom.utilcalculator.domain.tree.CommandTree;
 
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 public class UserProgress implements Serializable {
-    private TransportType transportType = null;
+    private GeneralTransportType generalTransportType = null;
     private CountryOrigin countryOrigin = null;
     private OwnersType ownersType = null;
     private CarAge carAge = null;
-    private TypeOfEngine typeOfEngine = null;
-    private VolumeOfEngine volumeOfEngine = null;
+    private EngineType engineType = null;
+    private EngineVolume engineVolume = null;
+    private BusesAndTrucksTransportType busesOrTrucksType = null;
+    private TruckUnitClass truckUnitClass = null;
+    private Weight weight = null;
+    private TrailerO4Type trailersO4Type = null;
     private final String chatID;
-    private Step currentQuestion;
+    private final List<Command> userPath;
 
     public UserProgress(final String chatID) {
         this.chatID = chatID;
+        userPath = new ArrayList<>();
+    }
+
+    private void addUserStatusToPath(final StepsIndicator stepsIndicator) {
+        final Map<StepsIndicator, Command> fieldsToCommands = CommandTree.init().getFieldsToCommands();
```

Comment:
These 3 lines should be enough I believe.
```suggestion
        final Map<StepsIndicator, Command> fieldsToCommands = CommandTree.init().getFieldsToCommands();
        Command command = fieldsToCommands.get(stepsIndicator);
        userPath.add( command);
```

Reply:
Sorry, don't understand your proposition. Probably, I should explain what's going on in this code.